globe-and-citizen/layer8-interceptor
# [DRAFT] Url path extraction and encryption at the User side

NOTE: Still need to do some tests with queries in the URL parameter.

## PR Description
Made changes at interceptor to extracr the URL path from the host (SP Backend URL), and move that path into the body and encrypt it, which will go all the way to the middleware.

Issue: https://github.com/globe-and-citizen/layer8/issues/65

## Manual Test Suite
SECTION 1: WEVE GOT POEMS (http://localhost:5173/)
[ ] Log in with 'tester' / '1234'
[ ] Log in anonymously with Layer8
[ ] Clicking "Get Next Poem" loads different poem correctly x 3
[ ] Clicking "Logout" takes you to the login screen
[ ] Clicking "Register" takes you to the registration page
[ ] Registering with a username, password, and profile image is successful
[ ] Logging in with the new username / password succeeds
[ ] Logging in with Layer8 opens the pop-up
[ ] Logging in with the "tester" & "12341234" works
[ ] User chooses to share their new "Username" & "Country" from the Layer8 Resource Server
[ ] Clicking "Get Next Poem" loads different poem correctly x 3
[ ] Clicking "Logout" takes you to the login page

SECTION 2: IMSHARER (http://localhost:5174/)
[ ] Main page loads
[ ] Upload of image works
[ ] Reload leads to instant reload (demonstrating proper caching)
[ ] Clicking the newly loaded image shows it in a light boxLog in using tester and 12341234 should succeed
